uberenv.py

```diff
@@ -494,13 +523,23 @@ class SpackEnv(UberEnv):
 
     def __init__(self, opts, extra_opts):
         UberEnv.__init__(self,opts,extra_opts)
-
         self.pkg_version = self.set_from_json("package_version")
-        self.pkg_final_phase = self.set_from_args_or_json("package_final_phase")
         self.pkg_src_dir = self.set_from_args_or_json("package_source_dir")
-
+        self.pkg_final_phase = self.set_from_args_or_json("package_final_phase",True)
+        # get build mode
+        self.build_mode = self.set_from_args_or_json("build_mode",True)
+        # default spack build mode is dev-build
+        if self.build_mode is None:
+            self.build_mode = "dev-build"
+        # NOTE: install always overrides the build mode to "install"
+        if self.opts["install"]:
+            self.build_mode = "install"
+        # if we are using fake package mode, adjust the pkg name
+        if self.build_mode == "uberenv-pkg":
+            self.pkg_name =  "uberenv-" + self.pkg_name
```

Comment:
this seems to be project specific. Can package_name/package-name not be "uberenv-conduit" in your case?

Reply:
I did have the exact name in there before, but that was the source of the confusing options name :-(

Reply:
we talked .. we decided to keep this behaviour.